src/java.desktop/share/native/libharfbuzz/hb-coretext.cc

```diff
@@ -190,13 +190,15 @@ create_ct_font (CGFontRef cg_font, CGFloat font_size)
    * reconfiguring the cascade list causes CoreText crashes. For details, see
    * crbug.com/549610 */
   // 0x00070000 stands for "kCTVersionNumber10_10", see CoreText.h
+#if TARGET_OS_OSX && MAC_OS_X_VERSION_MIN_REQUIRED < __MAC_10_10
```

Comment:
I need a robust explanation of these changes.
They are not mentioned, nor are they in upstream harfbuzz.
Likely these should be pushed to upstream first if there is a reason for them.

Reply:
I'm afraid it's one of those dirty changes that we did to make progress, but forgot to revisit later. Without the guard you would get:
```
* For target support_native_java.desktop_libharfbuzz_hb-coretext.o:
/Users/lewurm/work/aarch64-port/src/java.desktop/share/native/libharfbuzz/hb-coretext.cc:193:8: error: 'CTGetCoreTextVersion' is deprecated: first deprecated in macOS 11.0 - Use -[NSProcessInfo operatingSystemVersion] [-Werror,-Wdeprecated-declarations]
  if (&CTGetCoreTextVersion != nullptr && CTGetCoreTextVersion() < 0x00070000) {
       ^
/Applications/Xcode12.3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/CoreText.framework/Headers/CoreText.h:61:10: note: 'CTGetCoreTextVersion' has been explicitly marked deprecated here
uint32_t CTGetCoreTextVersion( void ) CT_DEPRECATED("Use -[NSProcessInfo operatingSystemVersion]", macos(10.5, 11.0), ios(3.2, 14.0), watchos(2.0, 7.0), tvos(9.0, 14.0));
         ^
/Users/lewurm/work/aarch64-port/src/java.desktop/share/native/libharfbuzz/hb-coretext.cc:193:43: error: 'CTGetCoreTextVersion' is deprecated: first deprecated in macOS 11.0 - Use -[NSProcessInfo operatingSystemVersion] [-Werror,-Wdeprecated-declarations]
  if (&CTGetCoreTextVersion != nullptr && CTGetCoreTextVersion() < 0x00070000) {
                                          ^
/Applications/Xcode12.3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/CoreText.framework/Headers/CoreText.h:61:10: note: 'CTGetCoreTextVersion' has been explicitly marked deprecated here
uint32_t CTGetCoreTextVersion( void ) CT_DEPRECATED("Use -[NSProcessInfo operatingSystemVersion]", macos(10.5, 11.0), ios(3.2, 14.0), watchos(2.0, 7.0), tvos(9.0, 14.0));
         ^
2 errors generated.
```
For now, it's probably better to go with what Anton did in the latest commit https://github.com/openjdk/jdk/pull/2200/commits/b2b396fca679fbc7ee78fb5bc80191bc79e1c490
Eventually upstream will do something about it I assume? How often does it get synced with upstream? Maybe worth to file an issue.